File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    arr1 = [1,2,3,4,5]
    arr2 = [2,1,2,3,2,4,2,5]
    arr3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    score = [0, 0, 0]
    for i, a in enumerate(answers):
        if a == arr1[i%len(arr1)]:
            score[0] += 1
        if a == arr2[i%len(arr2)]:
            score[1] += 1
        if a == arr3[i%len(arr3)]:
            score[2] += 1
    _max = max(score)
    for i, s in enumerate(score):
        if s == _max:
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[][] pattern = {
            {1, 2, 3, 4, 5},
            {2, 1, 2, 3, 2, 4, 2, 5},
            {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
        };
        int[] scores = new int[3];
        
        for(int i = 0 ; i < answers.length; i++){
            for(int j = 0; j < pattern.length; j++){
                if(answers[i] == pattern[j][i % pattern[j].length])
                    scores[j]++;
            }
        }
        int maxScore = 0;
        for(int i = 0; i < scores.length; i++){
            if(scores[i] > maxScore) maxScore = scores[i];
        }
        //int maxScore = Arrays.stream(scores).max().getAsInt();
        
        ArrayList<Integer> answer = new ArrayList<>();
        for(int i = 0; i < scores.length; i++){
            if(maxScore == scores[i])
                answer.add(i + 1);
        }
        int[] result = new int[answer.size()];
        for(int i = 0; i < answer.size(); i++){
            result[i] = answer.get(i);
        }
        return result;
        //return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}